packages/server/src/jsRunner.ts

```diff
@@ -1,61 +1,145 @@
-import vm from "vm"
+import ivm from "isolated-vm"
 import env from "./environment"
-import { setJSRunner } from "@budibase/string-templates"
-import { context, timers } from "@budibase/backend-core"
+import { setJSRunner, JsErrorTimeout } from "@budibase/string-templates"
+import { context } from "@budibase/backend-core"
 import tracer from "dd-trace"
+import fs from "fs"
+import url from "url"
+import crypto from "crypto"
 
-type TrackerFn = <T>(f: () => T) => T
+const helpersSource = fs.readFileSync(
+  `${require.resolve("@budibase/string-templates/index-helpers")}`,
+  "utf8"
+)
+
+class ExecutionTimeoutError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = "ExecutionTimeoutError"
+  }
+}
 
 export function init() {
-  setJSRunner((js: string, ctx: vm.Context) => {
+  setJSRunner((js: string, ctx: Record<string, any>) => {
     return tracer.trace("runJS", {}, span => {
-      const perRequestLimit = env.JS_PER_REQUEST_TIME_LIMIT_MS
-      let track: TrackerFn = f => f()
-      if (perRequestLimit) {
-        const bbCtx = tracer.trace("runJS.getCurrentContext", {}, span =>
-          context.getCurrentContext()
-        )
-        if (bbCtx) {
-          if (!bbCtx.jsExecutionTracker) {
-            span?.addTags({
-              createdExecutionTracker: true,
-            })
-            bbCtx.jsExecutionTracker = tracer.trace(
-              "runJS.createExecutionTimeTracker",
-              {},
-              span => timers.ExecutionTimeTracker.withLimit(perRequestLimit)
+      try {
+        const bbCtx = context.getCurrentContext()!
+
+        const isolateRefs = bbCtx.isolateRefs
+        if (!isolateRefs) {
+          const jsIsolate = new ivm.Isolate({ memoryLimit: 64 })
```

Comment:
We should make the `memoryLimit` configurable with an environment variable. I'm not sure what the right value is just yet, and self-hosters may want to significantly increase this.

Reply:
Thinking a bit more about this, it likely makes the most sense for this to be off by default (no limit), so as to preserve existing behaviour for self-hosters. Then we can tinker with it in cloud to find the best balance of protection for other workloads and not annoying customers.